xarray/backends/scipy_.py

```diff
@@ -63,8 +63,15 @@ def get_variable(self, needs_lock=True):
         ds = self.datastore._manager.acquire(needs_lock)
         return ds.variables[self.variable_name]
 
+    def _getitem(self, key):
+        with self.datastore.lock:
+            data = self.get_variable(needs_lock=False).data
+            return data[key]
+
     def __getitem__(self, key):
-        data = NumpyIndexingAdapter(self.get_variable().data)[key]
+        data = indexing.explicit_indexing_adapter(
```

Comment:
This doesn't seem right to me. The previous version was `NumpyIndexingAdapter` which supports vectorized indexing?

Reply:
good catch... i updated it to `indexing.IndexingSupport.OUTER_1VECTOR`

Reply:
Did you verify that `OUTER_1VECTOR` is correct for scipy netCDF3?

Reply:
yep, 

```python
In [2]: from xarray.tests import create_test_data
from 

In [4]: ds = create_test_data()

In [5]: ds.to_netcdf('/tmp/testing.nc', engine='scipy', format='NETCDF3_CLASSIC')

In [6]: dset = xr.open_dataset('/tmp/testing.nc', engine='scipy')
```

```python
In [7]: indexers = {"dim1": [1, 2, 0], "dim2": [3, 2, 0, 3], "dim3": np.arange(5)}

In [8]: dset.isel(**indexers).load()
Out[8]: 
<xarray.Dataset> Size: 564B
Dimensions:  (time: 20, dim3: 5, dim1: 3, dim2: 4)
Coordinates:
  * time     (time) datetime64[ns] 160B 2000-01-01 2000-01-02 ... 2000-01-20
  * dim3     (dim3) object 40B 'a' 'b' 'c' 'd' 'e'
    numbers  (dim3) int32 20B 0 1 2 0 0
  * dim2     (dim2) float64 32B 1.5 1.0 0.0 1.5
Dimensions without coordinates: dim1
Data variables:
    var3     (dim3, dim1) float64 120B 0.5684 1.263 1.24 ... -1.068 -1.538
    var1     (dim1, dim2) float64 96B -0.3767 -2.044 -1.223 ... 1.461 -2.978
    var2     (dim1, dim2) float64 96B -1.762 -0.1133 -0.6662 ... -0.4204 0.2655

In [9]: indexers = {"dim1" : xr.DataArray([0, 2, 0], dims="a"), "dim2": xr.DataArray([0
   ...: , 2, 3], dims="a")}

In [10]: dset.isel(**indexers)
Out[10]: 
<xarray.Dataset> Size: 592B
Dimensions:  (time: 20, dim3: 10, a: 3)
Coordinates:
  * time     (time) datetime64[ns] 160B 2000-01-01 2000-01-02 ... 2000-01-20
  * dim3     (dim3) object 80B 'a' 'b' 'c' 'd' 'e' 'f' 'g' 'h' 'i' 'j'
    numbers  (dim3) int32 40B ...
    dim2     (a) float64 24B 0.0 1.0 1.5
Dimensions without coordinates: a
Data variables:
    var3     (dim3, a) float64 240B ...
    var1     (a) float64 24B ...
    var2     (a) float64 24B ...

In [11]: dset.isel(**indexers).load()
Out[11]: 
<xarray.Dataset> Size: 592B
Dimensions:  (time: 20, dim3: 10, a: 3)
Coordinates:
  * time     (time) datetime64[ns] 160B 2000-01-01 2000-01-02 ... 2000-01-20
  * dim3     (dim3) object 80B 'a' 'b' 'c' 'd' 'e' 'f' 'g' 'h' 'i' 'j'
    numbers  (dim3) int32 40B 0 1 2 0 0 1 1 2 2 3
    dim2     (a) float64 24B 0.0 1.0 1.5
Dimensions without coordinates: a
Data variables:
    var3     (dim3, a) float64 240B 1.24 1.263 1.24 ... -1.465 0.552 -1.465
    var1     (a) float64 24B 1.461 -1.295 -2.978
    var2     (a) float64 24B -0.4204 -0.9256 0.2655

```

Reply:
Confirmed thanks @andersy005